R/utils.R

```diff
@@ -81,3 +56,57 @@ make_c_call <- function(choices) {
   if (length(private$session_bindings) > 0) lapply(private$session_bindings, function(x) x$destroy())
   invisible(NULL)
 }
+
+
+
+#' Encodes ids to be used in JavaScript and Shiny
+#'
+#' @description
+#' Replaces non-ASCII characters into a format that can be used in HTML,
+#' JavaScript and Shiny.
+#'
+#' When the id has a character that is not allowed, it is replaced with `"_"`
+#' and a hash of the original id is added to the beginning of the id.
+#'
+#' @param id (`character(1)`) The id string.
+#'
+#' @return Sanitized string that removes special characters and spaces.
+#'
+#' @keywords internal
+sanitize_id <- function(id) {
+  id_converted <- make.names(id)
+  if (identical(make.names(id), id)) {
```

Comment:
I agree, that's why two lines below it replaces `.` with `_`

I dropped that specific list as it will create a R-safe name.

Reply:
Check the latest commit, which adds some tests via `NS` call

Reply:
Yes, but look, if your `id = <dataname>` contains dots, it might be considered as valid and it will be returned as id (`return(id)`). It means that if dataset is named `my.data` - the id will be `<namespace>-my.data` which will cause problems

```
make.names("not.good.for.shiny.id")
# "not.good.for.shiny.id"
```

Reply:
Ahh, good point.

Reverted back to pattern conversion.